File: src/CSharp/601-700/669.TrimBST.cs

using System;
using System.Collections.Generic;
using System.Text;

namespace PeefyLeetCode.TrimBST
{
    public class TreeNode
    {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int x) { val = x; }
    }

public class Solution {
    public TreeNode TrimBST(TreeNode root, int L, int R) {
        if(root == null) return null;
        if(root.val < L) return TrimBST(root.right, L, R);
        if(root.val > R) return TrimBST(root.left, L, R);
        root.left = TrimBST(root.left, L, R);
        root.right = TrimBST(root.right, L, R);
        return root;
    }
}
}

File: src/Python/601-700/669.TrimBST.py

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def trimBST(self, root, L, R):
        """
        :type root: TreeNode
        :type L: int
        :type R: int
        :rtype: TreeNode
        """
        if root == None:
            return None
        if root.val >= L and root.val <= R:
            root.left = self.trimBST(root.left, L, R)
            root.right = self.trimBST(root.right, L, R)
        elif root.val < L:
            root = self.trimBST(root.right, L, R)
        elif root.val > R:
            root = self.trimBST(root.left, L, R)
        return root

if __name__ == '__main__':
    solution = Solution()
    root = TreeNode(1)
    root.left = TreeNode(0)
    root.right = TreeNode(2)
    root.right.right = TreeNode(4)
    print(solution.trimBST(root, 1, 2))
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left.right = TreeNode(4)
    print(solution.trimBST(root, 2, 3))
else:
    pass
